Add `klogctl` to linux and android

sgasse opened the pull request:
Those are useful to read the kernel message buffer.

References:
- https://android.googlesource.com/platform/bionic/+/master/libc/include/sys/klog.h
- https://man7.org/linux/man-pages/man2/syslog.2.html

sgasse commented:
@rustbot author

sgasse commented:
@rustbot review

tgross35 commented on src/unix/linux_like/linux/mod.rs:
```diff
@@ -5620,6 +5620,8 @@ extern "C" {
         flags: ::c_uint,
     ) -> ::ssize_t;
+
+    pub fn klogctl(syslog_type: ::c_int, buf: *mut ::c_char, len: ::c_int) -> ::c_int;
```
Nit: Linux docs call the second arg `bufp` per https://linux.die.net/man/3/klogctl

tgross35 commented:
Could you target `main` please? Otherwise lgtm, with one nit above.

@rustbot label +stable-nominated
@rustbot author

bors commented:
:umbrella: The latest upstream changes made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

sgasse commented:
Thank you for the review @tgross35 ! I updated the argument name as suggested in your review. Next I changed the base of the PR to `main`, rebased locally on `upstream/main` and pushed it (hopefully not breaking anything and not generating noise due to the base change).

sgasse commented:
@rustbot review

tgross35 reviewed (approved):
Thanks!
